joblib/hashing.py

```diff
@@ -146,7 +152,15 @@ def __init__(self, hash_name='md5', coerce_mmap=False):
         else:
             self._getbuffer = memoryview
 
-    def save(self, obj):
+    def persistent_id(self, obj):
+        # Separate dtypes shared by objects in the pickle graph,
+        # to avoid cache invalidation when unpickling a subobject of the graph
+        if isinstance(obj, self.np.dtype):
+            return pickle.dumps(obj)
+        else:
+            return None
+
+    def save(self, obj, save_persistent_id=True):
```

Comment:
Ok I get it, it's required by python 3. I don't get it why it was not required before, but fair enough.